### PR TITLE
Include virtual function overrides in constructor context.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/CallGraph.cpp [hunk 1]
```diff
@@ -55,7 +55,12 @@ bool CallGraph::visit(Identifier const& _identifier)
 {
 	FunctionDefinition const* fun = dynamic_cast<FunctionDefinition const*>(_identifier.getReferencedDeclaration());
 	if (fun)
+	{
+		if (m_overrideResolver)
+			fun = (*m_overrideResolver)(fun->getName());
+		solAssert(fun, "");
```
Comment (LefterisJP) [inline libsolidity/CallGraph.cpp]: Shouldn't we have a message in solAssert here?

File: libsolidity/Parser.cpp [hunk 2]
```diff
@@ -210,7 +210,8 @@ ASTPointer<FunctionDefinition> Parser::parseFunctionDefinition(bool _isPublic)
 	}
 	ASTPointer<Block> block = parseBlock();
 	nodeFactory.setEndPositionFromNode(block);
-	return nodeFactory.createNode<FunctionDefinition>(name, _isPublic, docstring,
+	bool const c_isConstructor = (_contractName && *name == *_contractName);
```
Comment (LefterisJP) [inline libsolidity/Parser.cpp]: why constant bool here?

File: libsolidity/AST.h [hunk 3]
```diff
@@ -280,13 +280,13 @@ class FunctionDefinition: public Declaration
 {
 public:
 	FunctionDefinition(Location const& _location, ASTPointer<ASTString> const& _name,
-					bool _isPublic,
+					bool _isPublic, bool _isConstructor,
```
Comment (LefterisJP) [inline libsolidity/AST.h]: Just to be consistent with the rest of the function, you could put `bool _isConstructor` to its own line to have all arguments on their own line

File: libsolidity/Parser.cpp (hunk 2 above)
Comment (chriseth) [inline libsolidity/Parser.cpp]: Why not?

Comment (LefterisJP) [inline libsolidity/Parser.cpp]: I think it's an overkill

**Pull request merged**